app/src/processing/app/Base.java

```diff
@@ -1877,6 +1877,15 @@ public void handleFontSizeChange(int change) {
     PreferencesData.set("editor.font", StringUtils.join(pieces, ','));
     getEditors().forEach(Editor::applyPreferences);
   }
+  
+  public void toggleLineNumber() {
+    if (PreferencesData.getBoolean("editor.linenumbers")) {
```

Comment:
Could be shortened to:
`PreferencesData.setBoolean("editor.linenumbers", !PreferencesData.getBoolean("editor.linenumbers"));`